providers/os/id/azcompute/azcompute.go

```diff
@@ -57,7 +57,7 @@ func (m *commandInstanceMetadata) Identify() (Identity, error) {
 	var instanceDocument string
 	switch {
 	case m.platform.IsFamily(inventory.FAMILY_UNIX):
-		cmd, err := m.conn.RunCommand("curl --noproxy '*' -H Metadata:true " + identityUrl)
+		cmd, err := m.conn.RunCommand("curl --max-time 1 --noproxy '*' -H Metadata:true " + identityUrl)
```

Comment:
@chris-rock seems fine to me too! I've set the timeout to 1 sec. because you did so at the PowerShell version of the metadata lookup.

https://github.com/mondoohq/cnquery/blob/9e1b2eb0c17e2045617c9350bccf1c102e4c672b/providers/os/id/azcompute/azcompute.go#L20

Shall we change both commands to the same values?

Reply:
@chris-rock which version of PowerShell does cnspec require? Options for "retry" and "delay" have been implement in PowerShell 7, most Windows systems still use PowerShell 5 at the moment.

If we wanna be backward compatible, we can increase timeout to 5 seconds without retries only.

Reply:
Lets stick with Powershell 5 for now. I think we should stick with 5 seconds for Pwsh, otherwise we hang on so long. In normal cases the metadata api should always respond much faster. So I think we can use shorter times just specifically for the metadata api. I still tend to stick with --max-time 5 since the payload from the metadata api is so small. WDYT?

Reply:
@chris-rock for me timeouts are relevent for the "worst case" - aren't they?

If the Metadata API responds faster, the command goes on right after getting a response.

The timeout is relevant in edge cases only - i.e. Linux VMs on Hyper-V.

For a bit better performance we could adjust **connection timeout** to 5 seconds for sure - I will take care of it.

Reply:
@chris-rock I've adjusted the timeouts to 5 seconds and removed the `--connect-timeout` from `curl`. Feel free to merge. :)

Reply:
@chris-rock after testing out several `curl` parameters, I think we should go with `curl --retry 5 --retry-delay 1 --connect-timeout 1 --retry-max-time 5 --max-time 10 ...`. This ensure a maximum **connection timeout of 5 seonds** and a **maximum tranfer timeout of 10 seconds**.

![image](https://github.com/user-attachments/assets/ef940d3e-5af7-44ef-9389-8cfd6dea85b0)

Reply:
Awesome. Thank you @tomtrix